crates/ide-assists/src/handlers/add_missing_match_arms.rs

```diff
@@ -446,21 +447,23 @@ fn build_pat(
                 mod_path_to_ast(&module.find_use_path(db, ModuleDef::from(var), prefer_no_std)?);
 
             // FIXME: use HIR for this; it doesn't currently expose struct vs. tuple vs. unit variants though
-            let pat: ast::Pat = match var.source(db)?.value.kind() {
+            Some(match var.source(db)?.value.kind() {
                 ast::StructKind::Tuple(field_list) => {
                     let pats =
                         iter::repeat(make::wildcard_pat().into()).take(field_list.fields().count());
                     make::tuple_struct_pat(path, pats).into()
                 }
                 ast::StructKind::Record(field_list) => {
-                    let pats = field_list
-                        .fields()
-                        .map(|f| make::ext::simple_ident_pat(f.name().unwrap()).into());
+                    let pats = field_list.fields().map(|f| {
+                        make::ext::simple_ident_pat(
+                            f.name().expect("Record field must have a name"),
```

Comment:
This feels like it could panic in the following situation, but I'm not exactly sure if it will:

```rust
enum A {
    A,
    Missing { a: u32, : u32, c: u32 }
}

fn a() {
    let b = A::A;
    match b$0 {}
}
```

Reply:
Thank you for mentioning this, I checked it and it doesn't. Syntax tree for the missing field name shows : 

```
[crates/ide-assists/src/handlers/add_missing_match_arms.rs:458] &f = RecordField {
    syntax: RECORD_FIELD@40..43
      NAME@40..43
        IDENT@40..43 "u32"
    ,
```
how little the piece is it is still being recognized as NAME and something like `{ a:32 , , c: u32}` completely skips seeing a record field between a and c. So we are good.

Reply:
Would be good to add this as a test case though, we might change recovery in the future which could change this behavior.